### PR TITLE
RavenDB-13903: Prevent from query operations (delete and patch) from …

### PR DESCRIPTION
**Pull request by maximburyak**
…blocking on execution. We want to add the operation and let the task to start running without any waiting

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs [hunk 1]
```diff
@@ -529,8 +529,11 @@ private void TrafficWatchQuery(IndexQueryServerSide indexQuery)
                 Query = query.Query
             };
 
-            var task = Database.Operations.AddOperation(Database, indexName, operationType,
-                onProgress => operation(Database.QueryRunner, options, onProgress, token), operationId, details, token);
+            var task =
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: I think that continuation few lines below will be done on a wrong task, so context will be disposed too early. Ping me when available, we have similar places to this in other parts of the system. I want to understand what is the cause and if it should be treat globally or not.

Comment (ayende) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: Pretty sure that `Task.Run(Func<Task>)` will return a task that only completes when the inner task task

Comment (ppekrol) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: Ok, but it does suggest that AddOperation should do that internally for all executions of that method. Not only here? There are 20 usages of this.

Comment (ppekrol) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: I've moved `Task.Run` inside the `AddOperation`. Waiting for CI.

**Review by ayende — APPROVED** [review]

**Pull request merged**